packages/devtools-evm-hardhat/src/tasks/healthcheck/validate-rpcs.ts

```diff
@@ -0,0 +1,111 @@
+
+const TIMEOUT = 1000 // 1 second
+
+const logger = createLogger()
+
+interface TaskArguments {
+    timeout: number
+}
+
+const getProvider = async (rpcUrl: string, networkName: string, eid: number): Promise<BaseProvider> => {
+    let provider
+    if (rpcUrl.startsWith(HTTP_URL) || rpcUrl.startsWith(HTTPS_URL)) {
+        const providerFactory = createProviderFactory()
+
+        try {
+            provider = await providerFactory(eid)
+        } catch (error) {
+            logger.error(
+                `Error fetching provider for network ${networkName}: ${error instanceof Error ? error.message : 'An unknown error occurred'}`
+            )
+        }
+    } else if (rpcUrl.startsWith(WS_URL) || rpcUrl.startsWith(WSS_URL)) {
+        provider = new ethers.providers.WebSocketProvider(rpcUrl)
+    } else {
+        logger.error(`Unsupported RPC protocol in network: ${networkName}`)
+    }
+
+    return provider
+}
+
+const action: ActionType<TaskArguments> = async (taskArgs, hre) => {
+    printLogo()
+
+    const networks = hre.userConfig.networks || {}
+    const eidByNetworkName = getEidsByNetworkName(hre)
+
+    logger.info(`========== Validating RPC URLs for networks: ${Object.keys(eidByNetworkName)}`)
+
+    const networksWithInvalidRPCs: string[] = []
+
+    await Promise.all(
+        Object.entries(eidByNetworkName).map(async ([networkName, eid]) => {
+            if (!eid) {
+                logger.info(`No eid found for network ${networkName}, skipping`)
+                return
+            }
+            const rpcUrl = networks[networkName]?.[RPC_URL_KEY]
+            if (!rpcUrl) {
+                logger.info(`No RPC URL found for network ${networkName}, skipping`)
+                return
+            }
+
+            const provider: BaseProvider = await getProvider(rpcUrl, networkName, eid as number)
+            if (!provider) {
+                networksWithInvalidRPCs.push(networkName)
+                logger.error(`Error fetching provider for network: ${networkName}`)
+                return
+            }
+
+            return Promise.race([
+                provider.getBlockNumber(),
+                new Promise<void>((_, reject) => setTimeout(reject, taskArgs.timeout)),
+            ]).then(
+                (block) => {
+                    return !!block
+                },
+                () => {
+                    networksWithInvalidRPCs.push(networkName)
+                }
+            )
+        })
+    )
+
+    if (networksWithInvalidRPCs.length !== 0) {
+        logger.error(
+            `========== ${printBoolean(false)} RPC URL validation failed for network(s): ${networksWithInvalidRPCs.join(', ')}`
+        )
+    } else {
+        logger.info(`========== ${printBoolean(true)} All RPC URLs are valid!`)
+    }
+}
+task(
+    TASK_LZ_VALIDATE_RPCS,
+    'Validate RPC URLs in hardhat.config.ts. RPCs are only considered valid if they use the https or wss protocol and respond within the specified timeout.',
```

Comment:
I think we can have RPCs with other protocols, but chainlist seems to only support HTTPS and WSS